aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -534,7 +586,27 @@ module post_mint_reveal_nft::minting {
         let resource_signer = create_signer_with_capability(&nft_mint_config.signer_cap);
 
         while (amount > 0) {
-            let token_id = token::mint_token(&resource_signer, source_token.token_data_id, 1);
+            let token_name = source_token.source_token_name_base;
+            string::append_utf8(&mut token_name, b": ");
+            let num = u64_to_string(source_token.source_token_counter);
+            string::append(&mut token_name, num);
+
+            let token_data_id = create_tokendata(
+                &resource_signer,
+                source_token.source_collection_name,
+                token_name,
+                collection_config.token_description,
+                collection_config.token_maximum,
+                source_token.source_token_uri,
+                collection_config.royalty_payee_address,
+                collection_config.royalty_points_den,
+                collection_config.royalty_points_num,
+                collection_config.token_mutate_config,
+                vector<String>[utf8(BURNABLE_BY_OWNER)],
+                vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+                vector<String>[utf8(b"bool")],
+            );
+            let token_id = token::mint_token(&resource_signer, token_data_id, 1);
```

Comment:
Isn't this expensive as it needs to create a new entry for the new token data?

Reply:
yeah but what I heard from the proj is that they want different token data for each source token

Reply:
why do they want different token data